src/main/java/com/iridium/iridiumteams/gui/BlockValueGUI.java

```diff
@@ -0,0 +1,125 @@
+package com.iridium.iridiumteams.gui;
+
+import com.iridium.iridiumcore.dependencies.xseries.XMaterial;
+import com.iridium.iridiumcore.gui.PagedGUI;
+import com.iridium.iridiumcore.utils.ItemStackUtils;
+import com.iridium.iridiumcore.utils.StringUtils;
+import com.iridium.iridiumteams.IridiumTeams;
+import com.iridium.iridiumteams.configs.BlockValues;
+import com.iridium.iridiumteams.configs.inventories.NoItemGUI;
+import com.iridium.iridiumteams.database.IridiumUser;
+import com.iridium.iridiumteams.database.Team;
+
+import lombok.Getter;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.EntityType;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class BlockValueGUI<T extends Team, U extends IridiumUser<T>> extends PagedGUI<BlockValues.ValuableBlock> {
+
+    private final T team;
+    private final IridiumTeams<T, U> iridiumTeams;
+    @Getter
+    private final BlockValueType blockValueType;
+
+    public BlockValueGUI(T team, BlockValueType blockValueType, Inventory previousInventory, IridiumTeams<T, U> iridiumTeams) {
+        super(
+                1,
+                iridiumTeams.getInventories().blockValueGUI.size,
+                iridiumTeams.getInventories().blockValueGUI.background,
+                iridiumTeams.getInventories().previousPage,
+                iridiumTeams.getInventories().nextPage,
+                previousInventory,
+                iridiumTeams.getInventories().backButton
+        );
+        this.team = team;
+        this.iridiumTeams = iridiumTeams;
+        this.blockValueType = blockValueType;
+    }
+
+    @NotNull
+    @Override
+    public Inventory getInventory() {
+        int maxPages = getPageObjects().size() / (getSize() - 9);
+        if (getPageObjects().size() % (getSize() - 9) > 0) maxPages++;
+
+        String guiTitle = "";
+        if (blockValueType == BlockValueType.BLOCK) guiTitle = iridiumTeams.getMessages().blockName;
+        if (blockValueType == BlockValueType.SPAWNER) guiTitle = iridiumTeams.getMessages().spawnerName;
```

Comment:
This should probabbly call `blockValuesTypeSelectorGUI` from Inventores class no? could always have a 

`blockValuesTypeSelectorGUI` and a `spawnerValuesTypeSelectorGUI`

Reply:
i based this off the missions gui, which lets you pick the category of mission, for parity with v3, but im open to changing it

Reply:
its just then we might need a ``/is spawnervalues`` command

Reply:
> its just then we might need a /is spawnervalues command
How so? You have the /blockvalue command which opens blockValueGUI, then do exactly the same thing your doing now, but its reading from two different configs in the inventory.yml